components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -490,4 +581,704 @@ mod tests {
             true
         );
     }
+
+    fn test_none_with_ctx_and_extra<F, Input, Ret>(func: F)
+    where
+        F: Fn(&mut EvalContext, &RpnFnCallExtra, &Option<Input>) -> Result<Option<Ret>>,
+    {
+        let mut ctx = EvalContext::default();
+        let implicit_args = [ScalarValue::Int(Some(1))];
+        let ret_field_type: FieldType = FieldType::default();
+        let extra = RpnFnCallExtra {
+            ret_field_type: &ret_field_type,
+            implicit_args: &implicit_args,
+        };
+        let r = func(&mut ctx, &extra, &None).unwrap();
+        assert!(r.is_none());
+    }
+
+    fn test_none_with_ctx<F, Input, Ret>(func: F)
+    where
+        F: Fn(&mut EvalContext, &Option<Input>) -> Result<Option<Ret>>,
+    {
+        let mut ctx = EvalContext::default();
+        let r = func(&mut ctx, &None).unwrap();
+        assert!(r.is_none());
+    }
+
+    fn test_none_with_extra<F, Input, Ret>(func: F)
+    where
+        F: Fn(&RpnFnCallExtra, &Option<Input>) -> Result<Option<Ret>>,
+    {
+        let implicit_args = [ScalarValue::Int(Some(1))];
+        let ret_field_type: FieldType = FieldType::default();
+        let extra = RpnFnCallExtra {
+            ret_field_type: &ret_field_type,
+            implicit_args: &implicit_args,
+        };
+        let r = func(&extra, &None).unwrap();
+        assert!(r.is_none());
+    }
+
+    fn test_none_with_nothing<F, Input, Ret>(func: F)
+    where
+        F: Fn(&Option<Input>) -> Result<Option<Ret>>,
+    {
+        let r = func(&None).unwrap();
+        assert!(r.is_none());
+    }
+
+    fn make_ctx(
+        overflow_as_warning: bool,
+        truncate_as_warning: bool,
+        should_clip_to_zero: bool,
+    ) -> EvalContext {
+        let mut flag: Flag = Flag::default();
+        if overflow_as_warning {
+            flag |= Flag::OVERFLOW_AS_WARNING;
+        }
+        if truncate_as_warning {
+            flag |= Flag::TRUNCATE_AS_WARNING;
+        }
+        if should_clip_to_zero {
+            flag |= Flag::IN_INSERT_STMT;
+        }
+        let cfg = Arc::new(EvalConfig::from_flag(flag));
+        EvalContext::new(cfg)
+    }
+
+    fn make_implicit_args(in_union: bool) -> [ScalarValue; 1] {
+        if in_union {
+            [ScalarValue::Int(Some(1))]
+        } else {
+            [ScalarValue::Int(Some(0))]
```

Comment:
```suggestion
            []
```

Reply:
The return value is `[ScalarValue; 1]`, so we can not do it like this.